File: frontend/lib/features/insurance/domain/entities/insurance.dart
import 'package:equatable/equatable.dart';

class Insurance extends Equatable {
  final String id;
  final String coupleId;
  final String userId;
  final String name;
  final String? insurer;
  final String insuranceType;
  final int premiumAmount;
  final int? paymentDay;
  final String paymentCycle;
  final String? paymentMethodId;
  final String? categoryId;
  final String? startDate;
  final String? endDate;
  final String? memo;
  final bool isActive;
  final String visibility;
  final String? ownerId;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Insurance({
    required this.id,
    required this.coupleId,
    required this.userId,
    required this.name,
    this.insurer,
    required this.insuranceType,
    required this.premiumAmount,
    this.paymentDay,
    required this.paymentCycle,
    this.paymentMethodId,
    this.categoryId,
    this.startDate,
    this.endDate,
    this.memo,
    required this.isActive,
    required this.visibility,
    this.ownerId,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        coupleId,
        userId,
        name,
        insurer,
        insuranceType,
        premiumAmount,
        paymentDay,
        paymentCycle,
        paymentMethodId,
        categoryId,
        startDate,
        endDate,
        memo,
        isActive,
        visibility,
        ownerId,
        createdAt,
        updatedAt,
      ];
}

class InsuranceSummaryItem extends Equatable {
  final String id;
  final String name;
  final String insuranceType;
  final int premiumAmount;
  final String paymentCycle;
  final int? paymentDay;
  final bool isActive;

  const InsuranceSummaryItem({
    required this.id,
    required this.name,
    required this.insuranceType,
    required this.premiumAmount,
    required this.paymentCycle,
    this.paymentDay,
    required this.isActive,
  });

  @override
  List<Object?> get props => [
        id,
        name,
        insuranceType,
        premiumAmount,
        paymentCycle,
        paymentDay,
        isActive,
      ];
}

class InsuranceSummary extends Equatable {
  final int year;
  final int month;
  final int totalPremium;
  final int activeCount;
  final List<InsuranceSummaryItem> items;

  const InsuranceSummary({
    required this.year,
    required this.month,
    required this.totalPremium,
    required this.activeCount,
    required this.items,
  });

  @override
  List<Object?> get props => [year, month, totalPremium, activeCount, items];
}

File: backend/src/main/kotlin/com/budgetbook/insurance/domain/Insurance.kt
package com.budgetbook.insurance.domain

import com.budgetbook.auth.domain.User
import com.budgetbook.category.domain.Category
import com.budgetbook.common.entity.BaseTimeEntity
import com.budgetbook.common.entity.Visibility
import com.budgetbook.couple.domain.Couple
import com.budgetbook.paymentmethod.domain.PaymentMethod
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDate
import java.util.UUID

@Entity
@Table(name = "insurances")
class Insurance(
    @Id
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "couple_id", nullable = false)
    val couple: Couple,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Column(nullable = false, length = 100)
    var name: String,

    @Column(length = 100)
    var insurer: String? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "insurance_type", nullable = false, length = 30)
    var insuranceType: InsuranceType,

    @Column(name = "premium_amount", nullable = false)
    var premiumAmount: Long,

    @Column(name = "payment_day")
    var paymentDay: Int? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "payment_cycle", nullable = false, length = 20)
    var paymentCycle: PaymentCycle = PaymentCycle.MONTHLY,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "payment_method_id")
    var paymentMethod: PaymentMethod? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    var category: Category? = null,

    @Column(name = "start_date")
    var startDate: LocalDate? = null,

    @Column(name = "end_date")
    var endDate: LocalDate? = null,

    var memo: String? = null,

    @Column(name = "is_active", nullable = false)
    var isActive: Boolean = true,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    var visibility: Visibility = Visibility.SHARED,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id")
    var owner: User? = null
) : BaseTimeEntity()
